libkbfs/tlf_journal.go

```diff
@@ -809,6 +808,62 @@ func (j *tlfJournal) getJournalStatus() (TLFJournalStatus, error) {
 	}, nil
 }
 
+func (j *tlfJournal) getJournalStatus() (TLFJournalStatus, error) {
+	j.journalLock.RLock()
+	defer j.journalLock.RUnlock()
+	return j.getBaseJournalStatusLocked()
+}
+
+func (j *tlfJournal) getJournalStatusWithPaths(ctx context.Context,
+	config Config, mdOps journalMDOps, blocks *folderBlockOps) (
+	TLFJournalStatus, error) {
+	j.journalLock.RLock()
+	defer j.journalLock.RUnlock()
+	status, err := j.getBaseJournalStatusLocked()
+	if err != nil {
+		return TLFJournalStatus{}, err
+	}
+
+	// Make chains over the entire range to get the unflushed files.
+	// TODO: cache this chains object and update it when new revisions
+	// appear and when revisions are flushed, instead of rebuilding
+	// every time.
+	if status.RevisionEnd != MetadataRevisionUninitialized {
+		var irmds []ImmutableRootMetadata
+		// These GetRange calls end up taking journalLock for reading
```

Comment:
I think returning partial results is fine, right? This will be used for status info and notifications, so I don't think it's critical to get it completely right.

Thinking about it, if we use `mdOps`, we won't get partial results, but we might get MDs that have already been sent to the server, which seems fine to me. If the branch changes, though, then we'd get an empty range, so maybe in that case it's worth retrying a few times and then giving up.